infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -434,6 +445,11 @@ public synchronized void resetServer(CorfuPayloadMsg<SequencerTailsRecoveryMsg>
             streamTailToGlobalTailMap.putAll(streamTails);
         }
 
+        // Update firstConsecutiveEpoch if the bootstrap epoch is consecutive.
```

Comment:
This optimization is very specific, why don't you address the general case (i.e. track a range of epochs as opposed to just the last epoch change) ?

Every time the sequencers epoch changes, you can maintain a consecutive range [a, b] and then any epoch within [a, b] can be accepted.

Reply:
@Maithem Sorry for the confusion, this comment is wrong,  missed a "not", should be "if the bootstrap epoch is not consecutive". I've made the correction. 

This patch does track a range of epochs: the `firstConsecutiveEpoch` is `a` in your example, and the current sequencer epoch is `b`.  Any time the sequencer epoch changes, we update `firstConsecutiveEpoch` if the new epoch is not consecutive. If it's consecutive, we don't update `firstConsecutiveEpoch`. So in this way, the range [firstConsecutiveEpoch, sequencer epoch] is a consecutive range. Any transaction falls in this range can be accepted for further resolution.